### PR TITLE
[ROCM] add is_compiled_with_rocm api, test=develop

### PR DESCRIPTION
**Pull request by qili93**
Add new api of "is_compiled_with_rocm"

https://github.com/PaddlePaddle/Paddle/pull/33043

文档预览如下：

http://10.136.157.23:8090/documentation/docs/zh//faq/index_cn.html?reviewVersion=jenkins-doc-review-772

image

image

### CODE REVIEW

File: docs/api/paddle/is_compiled_with_rocm_cn.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+.. py:function:: paddle.is_compiled_with_rocm()
+
+
+
+
+检查 ``whl`` 包是否可以被用来在AMD或海光GPU(ROCm)上运行模型
+
+返回：支持GPU(ROCm)则为True，否则为False。
+
+返回：是否支持GPU(ROCm)的bool值
```
Comment (TCChenlong) [inline docs/api/paddle/is_compiled_with_rocm_cn.rst]: 13、15 写成一行吧，
返回：bool，支持GPU(ROCm)则为True，否则为False。

File: docs/api/paddle/is_compiled_with_rocm_cn.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
+.. _cn_api_fluid_is_compiled_with_rocm:
```
Comment (TCChenlong) [inline docs/api/paddle/is_compiled_with_rocm_cn.rst]: _cn_api_fluid_is_compiled_with_rocm -> _cn_api_paddle_is_compiled_with_rocm

File: docs/api/paddle/is_compiled_with_rocm_cn.rst (hunk 1 above)
Comment (qili93) [reply docs/api/paddle/is_compiled_with_rocm_cn.rst]: Done

File: docs/api/paddle/is_compiled_with_rocm_cn.rst (hunk 2 above)
Comment (qili93) [reply docs/api/paddle/is_compiled_with_rocm_cn.rst]: Done

**Review by TCChenlong — APPROVED** [review]
LGTM

**Pull request merged**